### ISSUE TITLE
Allow downstream users to suppress Save Optimizer warnings

### ISSUE THREAD

**Issue by jsrozner** [pytorch/pytorch#46405]
## 🚀 Feature
Enable downstream users of this library to suppress lr_scheduler save_state_warning. (Propose to add an argument to LambdaLR [torch/optim/lr_scheduler.py])

If set to true, the warnings.warn(SAVE_STATE_WARNING, user_warning) that prints "Please also save or load the state of the optimizer when saving or loading the scheduler."  will not be generated.

## Motivation
Various bugs / discussions exist because users of various libraries are confused by this warning. Huggingface recently pushed a change to catch and suppress this warning.

[Recent Pytorch discussion](https://discuss.pytorch.org/t/weird-warning-popping-out-regarding-learning-rate-scheduler/99293)

[Huggingface issue](https://github.com/huggingface/transformers/issues/7765)

[Huggingface solution to deal with "the annoying warning"](https://github.com/huggingface/transformers/pull/7401)

## Pitch
Propose to add an argument to LambdaLR [torch/optim/lr_scheduler.py](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py)
* state_dict(..., suppress_state_warning=False)
* load_state_dict(..., suppress_state_warning=False)

Default false preserves the warning for everyone, except those who explicitly choose to set the flag, presumably because they have appropriately saved the optimizer.

## Alternatives
Huggingface implemented a wrapper to catch and suppress the warning but this is fragile.

## Concerns
Maybe there's some plumbing that should be updated to use this new flag, but once we provide the option to use the flag, others can begin implementing on their own.

cc @vincentqb

**Comment by jsrozner** [comment]
PS, I would be willing to write the PR! (I wanted to confirm that this is a reasonable idea, first)

**Comment by vincentqb** [comment]
Thanks for opening an issue for this! Since the warning has been part of pytorch for a bit, we can now simply remove the warning, and add a short comment in the docstring reminding this. The reference pull request explaining this is #43352.  Do you want to open a pull request to do this? Thanks again!
